doc-surrealdb_versioned_docs/version-1.x/reference-guide/vector-search.mdx

```diff
@@ -0,0 +1,219 @@
+
+## Vector Search in SurrealDB
+
+<ThemedImage
+  alt="Google search for the word lead"
+  className="lead screenshot"
+  sources={{
+    light: useBaseUrl('img/VC.png'),
+    dark: useBaseUrl('img/VC.png'),
+  }}
+/>
+
+The Vector Search feature of SurrealDB will help you do more and dig deeper into your data. 
+
+For example, if you wanted to search for an actor or actress who looked like you from an extensive dataset of movie artists, you would first use an ML model to convert the artist's images and details into vector embeddings and then use SurrealQL to find the artist with the most resemblance to your face vector embeddings. The more characteristics you decide to include in your vector embeddings, the higher the dimensionality of your vector will be, potentially improving the accuracy of the matches but also increasing the complexity of the vector search.
+
+```surql
+CREATE actor:1 SET name = 'Actor 1', embedding = [0.1, 0.2, 0.3, 0.4];
+CREATE actor:2 SET name = 'Actor 2', embedding = [0.2, 0.1, 0.4, 0.3];
+CREATE actor:3 SET name = 'Actor 3', embedding = [0.4, 0.3, 0.2, 0.1];
+CREATE actor:4 SET name = 'Actor 4', embedding = [0.3, 0.4, 0.1, 0.2];
+```
+
+Output:
+```surql
+[[{ embedding: [0.1f, 0.2f, 0.3f, 0.4f], id: actor:1, name: 'Actor 1' }], 
+[{ embedding: [0.2f, 0.1f, 0.4f, 0.3f], id: actor:2, name: 'Actor 2' }], 
+[{ embedding: [0.4f, 0.3f, 0.2f, 0.1f], id: actor:3, name: 'Actor 3' }], 
+[{ embedding: [0.3f, 0.4f, 0.1f, 0.2f], id: actor:4, name: 'Actor 4' }]]
+```
+
+You can assume your facial features to be
+
+```surql
+LET $person_embedding = [0.15, 0.25, 0.35, 0.45];
+```
+
+You have the option of using 3 different approaches for performing Vector search. 
+
+- [Bruteforce](/docs/surrealdb/surrealql/operators#brute-force-method-since-130)
+- [Hierarchical navigable small world(HNSW)](/docs/surrealdb/surrealql/statements/define/indexes#hnsw-hierarchical-navigable-small-world-since-150)
+- [M-Tree](/docs/surrealdb/surrealql/statements/define/indexes#m-tree-index-since-130)
+
+As you want to perform a nearest neighbour search and not an exact search, you would typically use an index like [HNSW](/docs/surrealdb/surrealql/statements/define/indexes#hnsw-hierarchical-navigable-small-world-since-150) or [M-Tree](/docs/surrealdb/surrealql/statements/define/indexes#m-tree-index-since-130). 
+
+```surql
+DEFINE INDEX mt_pts ON actor FIELDS embedding MTREE DIMENSION 4 TYPE F32;
+SELECT id, name, vector::similarity::cosine(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|2|> $person_embedding;
+SELECT id, vector::distance::euclidean(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|1|> $person_embedding;
+```
```

Comment:
The MTREE index is defined with a distance. Vectors in it are sorted depending on it. Euclidean is the distance by default. But if we want to return the closest result with the cosine distance, then the index must be defined with Cosine.

```surql
DEFINE INDEX mt_pts ON actor FIELDS embedding MTREE DIMENSION 4 TYPE F32;
SELECT id, vector::distance::euclidean(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|2|> $person_embedding;
```

```surql
DEFINE INDEX mt_pts ON actor FIELDS embedding MTREE DIMENSION 4 DIST COSINE TYPE F32;
SELECT id, name, vector::similarity::cosine(embedding, $person_embedding) AS dist FROM actor WHERE embedding <|2|> $person_embedding;
```

Output: 

```surql
[
	{
		dist: 0.9979654098963515f,
		id: actor:1
	},
	{
		dist: 0.9409388150451313f,
		id: actor:2
	}
]
```

Reply:
Yes, you are right. Thank you for correcting this @emmanuel-keller .